packages/core/src/mesh/PrimitiveMesh.ts

```diff
@@ -47,6 +45,10 @@ export class PrimitiveMesh {
     const countReciprocal = 1.0 / count;
     const segmentsReciprocal = 1.0 / segments;
 
+    const positions: Vector3[] = [];
```

Comment:
use "const positions: Vector3[] =new Array(vertexCount);" instead of "const positions: Vector3[] = [];" can improve performance